src/library/trackcollection.cpp

```diff
@@ -510,4 +516,68 @@ StorageUnicodeFunctions::likeFunction(sqlite3_context *p,
                                     itrString, endString, E));
 }
 */
+//Private function
+
+bool TrackCollection::settingsPathStatus(){
+    struct stat sb; //this struct will hold the access info of the dir using struct here as below it has function of same name
+    stat(m_pConfig->getSettingsPath().toStdString().c_str(), &sb);  //give the status, m_pConfig->getSettingsPath() will give the
```

Comment:
Use toLocal8Bit() to convert QStrings for char* system functions. 

Unfortunately this fails on windows because it has not Utf8 support. 
So you should prefer Qt Functions which are able to wrap such issues.